File: Leetcode_重刷紀錄/4.Problems605. Can Place Flowers/CanPlaceFlowers.cpp
// 605. Can Place Flowers
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {      
        if(n ==0)return true;
        int flowerbedSize = flowerbed.size();
        if(!flowerbed[0] && (flowerbedSize< 2||!flowerbed[1]))flowerbed[0] = 1, n--;

        for (int i = 1; i < flowerbedSize-1; i += (flowerbed[i] ? 2 :1)) {
            if (!(flowerbed[i] || flowerbed[i - 1] || flowerbed[i + 1])) 
            flowerbed[i] = 1, n--;
        }
        if(!(flowerbed[flowerbedSize - 1]|| flowerbed[flowerbedSize - 2])) n--;

        return n <= 0;
    }
};

File: Leetcode_重刷紀錄/4.Problems605. Can Place Flowers/CanPlaceFlowers.py
# 605. Can Place Flowers
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0 
        if flowerbed[0] == 0 and ( len(flowerbed) <= 1  or flowerbed[1] == 0 ):
            count =1
            flowerbed[0] = 1
        i = 1
        while i < len(flowerbed)-1:
            if flowerbed[i] +flowerbed[i - 1] + flowerbed[i + 1] == 0:
                flowerbed[i] = 1
                count += 1
                i += 2
            else:  
                i += 1
        if flowerbed[len(flowerbed)-1] == 0 and  flowerbed[len(flowerbed)-2] == 0 :
            count +=1

        return count >= n   